### PR TITLE
Test Tree Generation and Cell Simulation

### PR DESCRIPTION
**Pull request by gordonkoehn**
Testing blind-coded scripts/ revisions of Tree Generation and Cell Simulation.

Might implement basic Snakemake rules along the way. 

This PR is supposed to catch errors and bugs.

### CODE REVIEW

File: scripts/cell_simulation.py [hunk 1]
```diff
@@ -71,8 +78,11 @@ def create_parser() -> argparse.Namespace:
     parser.add_argument(
         "--strategy",
         required=False,
-        choices=["UNIFORM_INCLUDE_ROOT", "UNIFORM_EXCLUDE_ROOT"],
-        help="Cell attachment strategy",
+        choices=["UIR", "UXR"],
+        help="Cell attachment strategy,"
+        " UIR: UNIFORM_INCLUDE_ROOT,"
+        " UXR: UNIFORM_EXCLUDE_ROOT,"
+        " default: UIR",
```
Comment (gordonkoehn) [inline scripts/cell_simulation.py:85]: Adapting `cell_simulation.py` to new file naming.

File: scripts/cell_simulation.py [hunk 2]
```diff
@@ -125,7 +135,7 @@ def run_sim(params: argparse.Namespace) -> None:
     # Get Tree information from TreeId
     ###############################
     # make tree id from tree path
-    true_tree_fp = params.true_tree_fp
+    true_tree_fp = Path(params.true_tree_fp)
```
Comment (gordonkoehn) [inline scripts/cell_simulation.py:138]: minor bug

File: scripts/make_tree.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def create_parser() -> argparse.Namespace:
         required=True,
         help="tree type: random, deep, star, first letter only",
         type=str,
-        options=[
+        choices=[
             "r",
             "d",
             "s",
```
Comment (gordonkoehn) [inline scripts/make_tree.py:74]: minor bug

File: scripts/make_tree.py [hunk 4]
```diff
@@ -88,7 +88,10 @@ def main() -> None:
         None
     """
 
+    # unpack the arguments
     args = create_parser()
+    # cast
+    nodes = int(args.n_nodes)
```
Comment (gordonkoehn) [inline scripts/make_tree.py:94]: minor bug

**Review by gordonkoehn — COMMENTED** [review]
Caught 3 minor bugs, now the tree generation and cell simulation runs fine with the new file naming. I created 3 snakemake rules, to generate true trees and mutation matrices. Work well with the new file names. 

Deleted the old snakemake rules. My prior understanding of snakemake was very porous :) This seems better -.. What do you think ?

File: workflows/Snakefile [hunk 5]
```diff
@@ -33,56 +34,60 @@ rule all:
...
-
-
-rule generate_mock_data:
-    """Generates mock data using wildcards defined in rule:parametrize_mock_data"""
+        script="../scripts/make_tree.py"
+    output:
+        tree="../data/{experiment}/trees/T_{tree_type}_{n_nodes}_{tree_seed}.json"
+    shell:
+        """
+        poetry run python {input.script} \
```
Comment (pawel-czyz) [inline workflows/Snakefile:45]: To make it work with miniconda, you can remove `poetry run` from the Snakefile and work *within* `poetry shell`, which is similar to miniconda's environment.

**Review by pawel-czyz — APPROVED** [review]
LGTM!

File: workflows/Snakefile (hunk 5 above)
Comment (gordonkoehn) [reply workflows/Snakefile:45]: I tried just now -  but didn't get it to work. My console I run snakemake from gets caught in the poetry shell.

**Pull request merged**